### PR TITLE
[elixir] fix: schedule dmap tests failing.

### PR DESCRIPTION
**Pull request by grejdi-mbta**

### CODE REVIEW

**Comment by grejdi-mbta** [comment]
@paulswartz Do you mind checking if these environment variables are set to something/anything in Github Actions? I don't think I can.

* CUBIC_DMAP_CONTROLLED_USER_API_KEY
* CUBIC_DMAP_PUBLIC_USER_API_KEY

I fixed some logic in this PR and I need them set to make sure it works.

**Comment by paulswartz** [comment]
> @paulswartz Do you mind checking if these environment variables are set to something/anything in Github Actions? I don't think I can.
> 
>     * CUBIC_DMAP_CONTROLLED_USER_API_KEY 
>     * CUBIC_DMAP_PUBLIC_USER_API_KEY

I don't see anything on https://github.com/mbta/data_platform/settings/secrets/actions.

**Comment by grejdi-mbta** [comment]
> I don't see anything on https://github.com/mbta/data_platform/settings/secrets/actions.

@paulswartz Do you mind setting them like so? Just so we have values in there:
```
CUBIC_DMAP_CONTROLLED_USER_API_KEY=controlled_api_key
CUBIC_DMAP_PUBLIC_USER_API_KEY=public_api_key
```

I will add a test that covers the missing keys.

File: ex_cubic_ingestion/test/ex_cubic_ingestion/workers/schedule_dmap_test.exs [hunk 1]
```diff
@@ -7,7 +7,9 @@ defmodule ExCubicIngestion.Workers.ScheduleDmapTest do
   alias ExCubicIngestion.Workers.ScheduleDmap
 
   describe "perform/1" do
-    test "run job with error as dmap config items are not defined" do
+    test "run job errors out, as dmap config items are not defined" do
```
Comment (grejdi-mbta) [inline ex_cubic_ingestion/test/ex_cubic_ingestion/workers/schedule_dmap_test.exs:10]: test for erroring out when one of the env variables is not set.

**Review by paulswartz — COMMENTED** [review]
It looks like the tests are still failing?

**Comment by grejdi-mbta** [comment]
> It looks like the tests are still failing?

Just making sure. Are all of these in https://github.com/mbta/data_platform/settings/secrets/actions ?

* CUBIC_DMAP_BASE_URL
* CUBIC_DMAP_CONTROLLED_USER_API_KEY
* CUBIC_DMAP_PUBLIC_USER_API_KEY

**Comment by paulswartz** [comment]
Screenshot 2024-01-03 at 12 44 09 PM

**Comment by grejdi-mbta** [comment]
Tests are passing on my local, so I'm a little stumped.
@paulswartz Do you checking that `CUBIC_DMAP_BASE_URL` has a value as well please?

**Comment by paulswartz** [comment]
> > I don't see anything on https://github.com/mbta/data_platform/settings/secrets/actions.
> 
> @paulswartz Do you mind setting them like so? Just so we have values in there:
> 
> ```
> CUBIC_DMAP_CONTROLLED_USER_API_KEY=controlled_api_key
> CUBIC_DMAP_PUBLIC_USER_API_KEY=public_api_key
> ```

`CUBIC_DMAP_BASE_URL` does not have a value.

**Review by paulswartz — COMMENTED** [review]
They're set (you should be able to see them at https://github.com/mbta/data_platform/settings/variables/actions) but the test are still failing.

**Comment by grejdi-mbta** [comment]
> you should be able to see them at https://github.com/mbta/data_platform/settings/variables/actions

I don't have access to that page. 404.

File: ex_cubic_ingestion/test/support/httpoison.ex [hunk 2]
```diff
@@ -48,8 +48,6 @@ defmodule MockHTTPoison do
 
   @spec get!(String.t()) :: HTTPoison.Response.t()
   def get!(url) do
-    dmap_base_url = Application.fetch_env!(:ex_cubic_ingestion, :dmap_base_url)
```
Comment (grejdi-mbta) [inline ex_cubic_ingestion/test/support/httpoison.ex:51]: fixes warning for unused variable.

**Comment by paulswartz** [comment]
So I think that the easier way is for you to add these to the `env` section of the workflow file, since they aren't secret. Alternatively, you'll need to pull the values out of the `vars` context: https://docs.github.com/en/actions/learn-github-actions/variables#defining-configuration-variables-for-multiple-workflows

File: ex_cubic_ingestion/lib/ex_cubic_ingestion/workers/schedule_dmap.ex [hunk 3]
```diff
@@ -18,18 +18,43 @@ defmodule ExCubicIngestion.Workers.ScheduleDmap do
...
+      Application.fetch_env(:ex_cubic_ingestion, :dmap_controlled_user_api_key)
 
-        :ok
+    maybe_public_user_api_key =
+      Application.fetch_env(:ex_cubic_ingestion, :dmap_public_user_api_key)
 
-      _error ->
-        {:error, "dmap_base_url or dmap_api_key empty, dmap will not be scheduled"}
-    end
+    insert_fetch_jobs(
```
Comment (grejdi-mbta) [inline ex_cubic_ingestion/lib/ex_cubic_ingestion/workers/schedule_dmap.ex:28]: rewired this a little. made it easier for me.

**Comment by grejdi-mbta** [comment]
> So I think that the easier way is for you to add these to the `env` section of the workflow file, since they aren't secret. Alternatively, you'll need to pull the values out of the `vars` context: https://docs.github.com/en/actions/learn-github-actions/variables#defining-configuration-variables-for-multiple-workflows

All set. I had a misunderstanding that if the secrets were defined, then they would be available as environment variables, similar to ECS. They're not set on the workflow file, and everything is working correctly.

**Review by paulswartz — APPROVED** [review]

**Pull request merged**